main.go

```diff
@@ -0,0 +1,197 @@
+	locoMocoIngredients = append(locoMocoIngredients, "egg")
+	locoMocoIngredients = append(locoMocoIngredients, "bread bun")
+	locoMocoIngredients = append(locoMocoIngredients, "mushrooms")
+
+	locoMocoRecipe := Recipe{"loco moco", locoMocoIngredients, 54}
+
+	patatasBravasIngredients := make([]string, 0)
+	patatasBravasIngredients = append(patatasBravasIngredients, "potato")
+	patatasBravasIngredients = append(patatasBravasIngredients, "tomato")
+	patatasBravasIngredients = append(patatasBravasIngredients, "olive oil")
+	patatasBravasIngredients = append(patatasBravasIngredients, "onion")
+	patatasBravasIngredients = append(patatasBravasIngredients, "garlic")
+	patatasBravasIngredients = append(patatasBravasIngredients, "paprika")
+
+	patatasBravasRecipe := Recipe{"patas bravas", patatasBravasIngredients, 80}
+
+	friedRiceIngredients := make([]string, 0)
+	friedRiceIngredients = append(friedRiceIngredients, "rice")
+	friedRiceIngredients = append(friedRiceIngredients, "soy sauce")
+	friedRiceIngredients = append(friedRiceIngredients, "egg")
+	friedRiceIngredients = append(friedRiceIngredients, "onion")
+	friedRiceIngredients = append(friedRiceIngredients, "pea")
+	friedRiceIngredients = append(friedRiceIngredients, "carrot")
+	friedRiceIngredients = append(friedRiceIngredients, "sesame oil")
+
+	friedRiceRecipe := Recipe{"fried rice", friedRiceIngredients, 40}
+
+	// Create an interface of all the created recipes
+	recipes := []interface{}{eloteRecipe, locoMocoRecipe, patatasBravasRecipe, friedRiceRecipe}
+	insertManyResult, err := collection.InsertMany(context.TODO(), recipes)
+	if err != nil {
+		log.Fatal("Something went wrong trying to insert the new documents: ", err)
+	}
+	fmt.Println(insertManyResult.InsertedIDs, " documents successfully inserted.")
+
+	/*
+	 * *** FIND DOCUMENTS ***
+	 *
+	 * Now that we have data in Atlas, we can read it. To retrieve all of
+	 * the data in a collection, we create a filter for recipes that take
+	 * less than 45 minutes to prepare and sort by name (ascending)
+	 */
+
+	var filter = bson.M{"prepTimeInMinutes": bson.M{"$lt": 45}}
+	options := options.Find()
+	options.SetSort(bson.D{{"name", 1}}) // Sort by `name` field ascending
+
+	cursor, err := collection.Find(context.TODO(), filter, options)
+	if err != nil {
+		log.Fatal("Something went wrong trying to find the documents: ", err)
+	}
+
+	// Loop through the returned recipes
+	for cursor.Next(ctx) {
+		// declare a result BSON object
+		var result bson.M
+		err := cursor.Decode(&result)
+		// If there is a cursor.Decode error
+		if err != nil {
+			log.Fatal("cursor.Next() error:", err)
+		} else {
+			var name = result["name"].(string)
+			var ingredients = result["ingredients"]
+			var prepTimeInMinutes = result["prepTimeInMinutes"].(int32)
+
+			fmt.Println(name, "has the ingredients:", ingredients, "and takes", prepTimeInMinutes, "minutes to make")
+		}
+	}
+
+	// We can also find a single document. Let's find the first document
```

Comment:
I don't mind that you're using a different query filter than the C# app, but I think part of the intent of searching on an array field (ingredients) is lost.

Reply:
I'm using the same query in the node.js app (https://github.com/mongodb-university/atlas_starter_nodejs/blob/master/app.js#L107), would you recommend keeping this filter + change the comment to be similar to the node app, or changing the filter to the same one C# uses (find all recipes that contain the ingredient "potato")

Reply:
The line you linked is not the same step.

Here's what you're looking for:
https://github.com/mongodb-university/atlas_starter_nodejs/blob/master/app.js#L122

```
 const findOneQuery = { ingredients: "potato" };
```
Your filter is:
```
var myFilter = bson.D{{"name", "fried rice"}}
```

Maybe you were working from an older version?

Reply:
whoops sorry about that, I've switched it to { ingredients: "potato" } now